airflow-core/src/airflow/api_fastapi/core_api/routes/public/extra_links.py

```diff
@@ -86,6 +86,22 @@ def get_extra_links(
         (link_name, task.get_extra_links(ti, link_name)) for link_name in task.extra_links
     )
     all_extra_links = {link_name: link_url or None for link_name, link_url in sorted(all_extra_link_pairs)}
+    # If no links found, or some are missing, try runtime (_link_*) keys from XCom
+    if not all_extra_links or any(link is None for link in all_extra_links.values()):
+        from airflow.models.xcom import XComModel
+
+        query = XComModel.get_many(
+            dag_ids=dag_id,
+            run_id=dag_run_id,
+            task_ids=task_id,
+            map_indexes=map_index,
+            session=session,
+        )
+        for row in query.all():
+            if row.key.startswith("_link_"):
```

Comment:
I'd rather we didn't get everything to then filter this in python space.

```suggestion
        for row in query.filter(XComModel.key.like(r"\_link\_"):
```

`\_` is because `_` is the wildcard character for any one character.